lib/stream/leo-stream.js

```diff
@@ -1275,207 +1278,208 @@ module.exports = function(configure) {
-						hasLoops--;
-						getEvents(function(err, items) {
-							if (err) {
-								return done(err);
-							}
-							logger.debug("found", items.length, "items");
-							if (items.length == 0) {
-								if (usingSnapshot) { //Then let's go back to the real queue
-									queue = originalQueue;
-									start = snapshotNext;
-									usingSnapshot = false;
-									return done();
-								} else if (usingArchive) { //Then let's go back to the real queue
-									queue = originalQueue;
-									usingArchive = false;
-									return done();
-								} else {
-									logger.debug("no more events");
-									hasMoreEvents = false;
-									return done();
-								}
-							}
-
-							var counts = 0;
-							var size = 0;
-							async.eachOfSeries(items, (item, i, done) => {
-								var cb = done;
-								done = () => {
-									process.nextTick(() => cb.apply(cb, arguments));
-								};
-								if (totalCount >= opts.limit || totalSize >= opts.size || !hasTime || isPassDestroyed) {
-									return done();
+							logger.debug("checking next loop, loops remaining ", hasLoops, ", Time Remaining:", opts.stopTime - moment.now());
+							hasTime = hasTime && (opts.stopTime > moment.now());
+							logger.debug(totalCount, opts.limit, totalSize, opts.size, hasMoreEvents, hasTime, hasLoops, max_eid, start, (usingSnapshot || max_eid.localeCompare(start)));
+							return totalCount < opts.limit && totalSize < opts.size && hasMoreEvents && hasTime && hasLoops && (usingSnapshot || max_eid.localeCompare(start) > 0);
+						}, (done) => {
+							var count = 0;
+							hasLoops--;
+							getEvents(function(err, items) {
+								if (err) {
+									return done(err);
 								}
-								if (item.start) {
-									var _parts = item.start.split(/-/);
-									var prefix = _parts[0];
-									var idOffset = parseInt(_parts[1]);
+								logger.debug("found", items.length, "items");
+								if (items.length == 0) {
+									if (usingSnapshot) { //Then let's go back to the real queue
+										queue = originalQueue;
+										start = snapshotNext;
+										usingSnapshot = false;
+										return done();
+									} else if (usingArchive) { //Then let's go back to the real queue
+										queue = originalQueue;
+										usingArchive = false;
+										return done();
+									} else {
+										logger.debug("no more events");
+										hasMoreEvents = false;
+										return done();
+									}
 								}
 
-								let createEId = null;
-								if (usingSnapshot) {
-									createEId = function(eid) {
-										return "_snapshot/" + prefix + "-" + (pad + (idOffset + eid)).slice(padLength);
-									};
-								} else {
-									createEId = function(eid) {
-										return prefix + "-" + (pad + (idOffset + eid)).slice(padLength);
-									};
-								}
+								var counts = 0;
+								var size = 0;
+								async.eachOfSeries(items, (item, i, done) => {
+										var cb = done;
+										done = (err) => {
```

Comment:
Are you wanting to do something with err? It's not being used here at the moment.